Add playbook configurations resource REST specification

mikreite opened the pull request:
I'd like to add the DoNotMerge label, but the UI would not allow me to do so.
Please advise.

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

mikreite committed:
add playbook configurations resource REST specification

AutorestCI commented:
# Automation for azure-sdk-for-ruby

AutorestCI commented:
# Automation for azure-sdk-for-python

AutorestCI commented:
# Automation for azure-sdk-for-java

AutorestCI commented:
# Automation for azure-sdk-for-js

AutorestCI commented:
# Automation for azure-sdk-for-go

adxsdknet commented:
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5688 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5688/commits/6054083b6b9269bc9fe136f390ae682624358efe

azuresdkci commented:
Can one of the admins verify this patch?

praries880 commented:
@mikreite Kindly ping me when this PR is ready to review.

chlahav commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json:
```diff
@@ -0,0 +1,461 @@
+        },
+        "tags": [ "PlaybookConfigurations" ],
+        "description": "Get all playbook configurations in a subscription.",
+        "operationId": "PlaybookConfigurations_ListBySubscription",
```
please change to "PlaybookConfigurations_List"

mikreite replied:
This name is already taken by another operationid (List resources under a given resource group)

chlahav commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json:
```diff
@@ -0,0 +1,461 @@
+        },
+        "tags": [ "PlaybookConfigurations" ],
+        "description": "Get the details of a playbook configuration.",
+        "operationId": "PlaybookConfigurations_GetByResourceGroup",
```
please change to "PlaybookConfigurations_Get"

chlahav commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json:
```diff
@@ -0,0 +1,461 @@
+        },
+        "tags": [ "PlaybookConfigurations" ],
+        "description": "Create or update a playbook configuration. If a playbook configuration is already created and a subsequent create request is issued with different properties, the playbook configuration properties will be updated.",
+        "operationId": "PlaybookConfigurations_CreateOrUpdateByResourceGroup",
```
"PlaybookConfigurations_CreateOrUpdate"

chlahav commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json:
```diff
@@ -0,0 +1,461 @@
+          {
+            "$ref": "#/parameters/PlaybookConfigurationName"
+          },
+          {
```
export to parameter definition - do not declare in here directly

mikreite replied:
done

chlahav commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json:
```diff
@@ -0,0 +1,461 @@
+          }
+        ],
+        "responses": {
+          "200": {
```
when will you have 200 and when 201?

mikreite replied:
201 - for a successful resource creation
200 - for a successful resource update

chlahav commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json:
```diff
@@ -0,0 +1,461 @@
+          }
+        },
+        "tags": [ "PlaybookConfigurations" ],
+        "operationId": "PlaybookConfigurations_DeleteByResourceGroup",
```
"PlaybookConfigurations_Delete"

chlahav commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json:
```diff
@@ -0,0 +1,461 @@
+        },
+        "scopes": {
+          "type": "array",
+          "description": "A collection of the resources scopes on which the playbook configurations logic is applied.",
```
add the scope types that are supported to the description

chlahav commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json:
```diff
@@ -0,0 +1,461 @@
+        "createdBy": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The AAD object id that created the playbook configuration."
```
```suggestion
          "description": "The AAD object ID of the entity that created the playbook configuration."
```

chlahav commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json:
```diff
@@ -0,0 +1,461 @@
+        "lastUpdatedBy": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The AAD object id that last updated the playbook configuration."
```
```suggestion
          "description": "The AAD object ID of the entity that last updated the playbook configuration."
```

chlahav commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json:
```diff
@@ -0,0 +1,461 @@
+      "type": "object",
+      "description": "The source event types which evaluate the playbook configuration set of rules. For example - security alerts and recommended tasks.",
+      "properties": {
+        "eventSource": {
```
if this is an enum add that to the swagger including all the possible values

mikreite replied:
for pp, it will be supplied via documentation. For PP, there will be a dedicated API for the valid values.

majastrz replied:
ok

---
In reply to: [276255184](https://github.com/Azure/azure-rest-api-specs/pull/5548#discussion_r276255184) [](ancestors = 276255184)

chlahav commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json:
```diff
@@ -0,0 +1,461 @@
+    },
+    "PlaybookConfigurationAction": {
+      "type": "string",
+      "description": "The uri that should be triggered.",
```
add properties or description of how it will be triggered (GET, POST etc.)

chlahav commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json:
```diff
@@ -0,0 +1,461 @@
+          "type": "string",
+          "description": "The expected value."
+        },
+        "conditions": {
```
```suggestion
        "condition": {
```

mikreite replied:
changed to "operator"

chlahav commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json:
```diff
@@ -0,0 +1,461 @@
+        },
+        "actions": {
+          "type": "array",
+          "description": "A collection of the actions which are triggered upon true evaluation of the configured rule sets.",
```
The important part here is the "all"

```suggestion
          "description": "A collection of the actions which are triggered if all the configured rule set evaluation is true.",
```

chlahav commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json:
```diff
@@ -0,0 +1,461 @@
+          "type": "array",
+          "description": "A collection of the actions which are triggered upon true evaluation of the configured rule sets.",
+          "items": {
+            "$ref": "#/definitions/PlaybookConfigurationAction"
```
since this is just a string you can collapse the definition to here for readability

mikreite replied:
apparently we are back to object here..

mikreite replied:
done

chlahav commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/PlaybookConfigurations/GetPlaybookConfigurationResourceGroup_example.json:
```diff
@@ -0,0 +1,52 @@
+                  "rules": [
+                    {
+                      "propertyJPath": "AlertType",
+                      "propertyType": "System.String",
```
please use a standard or your own enum
not all of our customers are .Net developers 😄

mikreite replied:
:) I was thinking of exposing a specific list via documentation, rather than an enum here to avoid continuous modifications (the same as in Compute [e.g.](https://github.com/Azure/azure-rest-api-specs/blob/6081866424e897b5ba7890bdb7279f4a02e481e6/specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/compute.json#L854) ). What's better?

chlahav commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/PlaybookConfigurations/GetPlaybookConfigurationResourceGroup_example.json:
```diff
@@ -0,0 +1,52 @@
+                {
+                  "rules": [
+                    {
+                      "propertyJPath": "AlertType",
```
for standard path you can use "$.type" or just "type"
it should not add work on your side since there are a lot of jpath evaluators

chlahav commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json:
```diff
@@ -0,0 +1,461 @@
+          "description": "The expected value."
+        },
+        "conditions": {
+          "type": "string",
```
add enum description

chlahav commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json:
```diff
@@ -0,0 +1,461 @@
+          "type": "boolean",
+          "description": "Indicates whether the model is valid or not."
+        },
+        "message": {
```
maybe you can add a list of all the failed validations?
this way the user will reduce the number of request count before a success

mikreite replied:
This validations model will be designed for the PP version. At the meantime, for pp, all validation errors will be concatenated to a single string. Is it ok?

majastrz replied:
It sounds like you're planning to return 200 OK with isValid set to false for invalid playbooks. You should return 200 OK only if the validation is successful. For validation failures, you should return 400 Bad Request and the error information should be captured inside the standard ARM error response. Take a look at "Error Response Content" section at [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md). 

---
In reply to: [273947475](https://github.com/Azure/azure-rest-api-specs/pull/5548#discussion_r273947475) [](ancestors = 273947475)

mikreite committed:
fix according to PR comments + some spec changes
introduce trackedResource entity

mikreite committed:
add propertyType and operator enums

mikreite commented:
Hi @praries880, the PR is ready to review

praries880 commented:
@mikreite  kindly take care of the merge conflict

mikreite committed:
merge from upstream/master

mikreite committed:
add 201 response to x-ms-example
fix the PlaybookConfigurationRuleSet definition
add a new reference to the readme.md file

mikreite committed:
remove the unnecessary "value" from the Put example response

mikreite commented:
Thanks @praries880 for the notification, merge conflict has been resolved

mikreite commented:
Hi @praries880 could you please add the "DoNotMerge" label back? We are on private preview with these APIs, thanks!

mikreite commented:
adding @dafishma for visibility

mikreite commented:
@ravbhatnagar appreciate your review, thanks

dsgouda commented:
startbuild

DFMSUser committed:
Merge remote-tracking branch 'upstream/master'

adxsdknet commented:
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5998 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5998/commits/ac7f4833a2d8b3a1b44cc20610ea00121afd7372

majastrz commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json:
```diff
@@ -0,0 +1,511 @@
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Security/playbookConfigurations": {
```
>playbookConfigurations [](start = 66, length = 22)

You should specify the x-ms-pageable extension ([https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-pageable](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-pageable)) with nextLinkName set to "nextLink".

dafishma replied:
resolved (added x-ms-pageable to all get collections)

majastrz commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json:
```diff
@@ -0,0 +1,511 @@
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Security/playbookConfigurations": {
```
>playbookConfigurations [](start = 101, length = 22)

You should specify the x-ms-pageable extension ([https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-pageable](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-pageable)) with nextLinkName set to "nextLink".

dafishma replied:
resolved (added x-ms-pageable to all get collections)

majastrz commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json:
```diff
@@ -0,0 +1,511 @@
+          }
+        }
+      },
+      "put": {
```
>put [](start = 7, length = 3)

Is this going to be a tracked resource in ARM. If yes, then you will also need to support PATCH with ability to patch resource tags at the minimum.

dafishma replied:
Done. Added patch api and example (for tags only)

majastrz commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json:
```diff
@@ -0,0 +1,511 @@
+          "type": "boolean",
+          "description": "Indicates whether the playbook configuration is enabled."
+        },
+        "metadata": {
```
>metadata [](start = 9, length = 8)

Should this be called "status" rather than "metadata"?

dafishma replied:
No, this is the metadata that describes the playbookConfiguration. Currently the meta-data describes who updated it and when (additional info will be added in the future)

majastrz replied:
ok

---
In reply to: [281738432](https://github.com/Azure/azure-rest-api-specs/pull/5548#discussion_r281738432) [](ancestors = 281738432)

majastrz commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json:
```diff
@@ -0,0 +1,511 @@
+            "System.Double",
+            "System.Char"
+
+          ]
```
The list of types is at odds with `propertyJPath`. By looking at this API, my expectation is that the JPath will be executed on something and the propertyType is expected to match. JSON supports the following data types: string, number, object, array, boolean, null. What will be the differences between byte, int16, int32, and double?

dafishma replied:
Done. Good call! And agreed. Changed this to support the standard Swagger/JSON data types:
"string", "integer", "number", "boolean"

majastrz commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json:
```diff
@@ -0,0 +1,511 @@
+        "resourceProviderNamespace": {
+          "type": "string",
+          "description": "The resource provider name."
+        },
```
What is the relationship between `resourceId` and `resourceProviderNamespace`. Can the namespace not be inferred from the resource ID?

dafishma replied:
This was changed. There will be future scenarios that need this information (for non-azure providers as well).

For now, this will include "logicAppResourceId", "resourceProviderNamespace" and "uri"

majastrz replied:
Trying to jam settings specific to Azure and non-Azure RPs into one object will result in a confusing API because you have to guess which properties are valid for which scenario. 

My suggestion would be to add a kind property and use it as a discriminator when you get a 2nd scenario. Then, per value of the kind property, you can have a completely different schema. Take a look at how Microsoft.Blueprint RP is doing it with their artifacts resources. The good news is that you can add that later without breaking changes in the API with what you have so far.

---
In reply to: [281745248](https://github.com/Azure/azure-rest-api-specs/pull/5548#discussion_r281745248) [](ancestors = 281745248)

majastrz replied:
Btw, my original point was that resourceProviderNamespace is redundant and can be inferred from the resourceId. In your examples, the Microsoft.Logic namespace is present in the resource ID. Why does the user have to specify it if we can just parse it out of the ID? Are there cases where the 2 namespaces are different?

---
In reply to: [282312303](https://github.com/Azure/azure-rest-api-specs/pull/5548#discussion_r282312303) [](ancestors = 282312303,281745248)

dafishma replied:
The resource provider namespace can indeed be inferred however these will not always be Azure Resource Providers. We can change the name if this is too confusing but the meaning here is the "provider of the selected resource", not necessarily the "azure provider". If this is confusing, we can change the name.

Can you elaborate on the use of "kind" property? 
Code-wise, these are not the same objects, but this is added to apply different set of required access for each different ID type.

dafishma replied:
Hi, we had additional thoughts on this - since we want several actions of different types and since we cant do this in the entire object, we added discriminator to the Action (actionType) and used this.

majastrz commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json:
```diff
@@ -0,0 +1,511 @@
+          "type": "string",
+          "description": "The resource provider name."
+        },
+        "uri": {
```
>uri [](start = 9, length = 3)

Are these going to be ARM calls or data plane calls? How will auth work here?

dafishma replied:
There are many different types. Can be both potentially. Auth is handled differently on different action types.

The first action type supported, (logic app), the auth is send as part of the URI, as defined by LogicApp.

majastrz replied:
Btw, if a property contains values that could be potentially considered a secret, you should not be returning it on GETs.

---
In reply to: [281746039](https://github.com/Azure/azure-rest-api-specs/pull/5548#discussion_r281746039) [](ancestors = 281746039)

dafishma replied:
The secrets are not returned in the GETs (a text is returned specifying that value is a secret and can only be updated, not obtained)

majastrz replied:
ok

---
In reply to: [282745636](https://github.com/Azure/azure-rest-api-specs/pull/5548#discussion_r282745636) [](ancestors = 282745636)

majastrz commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/PlaybookConfigurations/PutPlaybookConfiguration_example.json:
```diff
@@ -0,0 +1,175 @@
+                      "propertyJPath": "$.type",
+                      "propertyType": "System.String",
+                      "expectedValue": "customType",
+                      "operator": "EqualsIgnoreCase"
```
How does this deal with api versions? Resource payloads are allowed to change between API versions, which would invalidate the rules.

dafishma replied:
We have an internal version handling, that also uses the alias mechanism.

dafishma replied:
We are starting with support for JPaths (where versions are handled internally), and we will be adding future support for aliases as well (different scenarios).

majastrz replied:
That's great that you're reusing aliases, but the API suggests that you're operating on the resource body `$.type` looks a lot like querying the "type" property on a resource payload. How does this actually work? Can elaborate on how you handle versions internally?

---
In reply to: [281746903](https://github.com/Azure/azure-rest-api-specs/pull/5548#discussion_r281746903) [](ancestors = 281746903)

dafishma replied:
You are correct, it is a bit confusing. Our choice of example was not a good one. We updated it to reflect a different JPath.

Just to clarify, we are not querying the type at all (as explained, its a bad example on our part). The user specifies a JPath to an object, specifies the type to expect, the value to expect and an operator from a list of known operators (supplied via documentation).

In the future we will support also Aliases instead of specific JPaths.

majastrz commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json:
```diff
@@ -0,0 +1,511 @@
+          "description": "A valid event source type."
+        },
+        "ruleSets": {
+          "type": "array",
```
I noticed that there are a few array properties in your resource. Have you considered modeling them as a child resource instead? Would the user ever need to change them independently from the parent resource? Would a user ever have permissions to touch the child but not the parent or vice versa?

dafishma replied:
The parent cannot exist without its children in this case, which is why it is not supported.

majastrz commented on specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/PlaybookConfigurations/PutPlaybookConfiguration_example.json:
```diff
@@ -0,0 +1,175 @@
+              "resourceProviderNamespace": "Microsoft.Logic",
+              "uri": "https://triggerUri2.com"
+            }
+          ]
```
A lot of this looks very similar to how Azure policy declares policy definitions. Have you looked at Azure policy at all?

dafishma replied:
Yes of course. Much of this (and the language) was setup with Azure Policy (we are working on integration with them).

majastrz reviewed (changes requested):
I added some comments and questions. Please take a look.

DFMSUser committed:
Updating with latest PR

dafishma commented:
@majastrz - updated all of your comments. Please review them.

majastrz reviewed (changes requested):
Confirmed that comments were addressed. Can you take a look at the questions I added?

praries880 commented:
@mikreite  Kindly take care of the following model validation errors :https://dev.azure.com/azure-sdk/public/_build/results?buildId=27341&view=logs&jobId=8d2f9c49-cd83-5a9d-f3fe-2fd651afa742&taskId=43324f1f-6724-5d61-785f-f0ac9fd256a1&lineStart=34&lineEnd=35&colStart=1&colEnd=1

DFMSUser committed:
Fixing Example files

DFMSUser committed:
Adding descriminator and updating examples

dafishma commented:
@majastrz can you please confirm that all comments were addressed please?
Are we good to go?

jhendrixMSFT commented:
@AutorestCI regenerate azure-sdk-for-go

praries880 commented:
@mikreite Is the PR ready to merge now?

praries880 commented:
@mikreite pinging back

dafishma commented:
Hi,

The PR cannot be merged yet. We are still in a preview phase to check this.
Will update when it is ready to complete.

Thanks,
David Fishman

From: Pareekshit Singh <notifications@github.com>
Sent: Monday, July 1, 2019 9:24 PM
To: Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com>
Cc: David Fishman <dafishma@microsoft.com>; Mention <mention@noreply.github.com>
Subject: Re: [Azure/azure-rest-api-specs] Add playbook configurations resource REST specification (#5548)


@mikreite<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmikreite&data=02%7C01%7Cdafishma%40microsoft.com%7C08b62f7cb26e46c8fef308d6fe51438f%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636976022311942883&sdata=47A07NLUcZWzv%2BcxWFXPqOlWp94MUFLit6sqPYGMm00%3D&reserved=0> Is the PR ready to merge now?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F5548%3Femail_source%3Dnotifications%26email_token%3DAL4WNRJZ4W4YJYLNAB45QWLP5JDTLA5CNFSM4HDUI7F2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODY66JTY%23issuecomment-507372751&data=02%7C01%7Cdafishma%40microsoft.com%7C08b62f7cb26e46c8fef308d6fe51438f%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636976022311942883&sdata=tqQ8Rh6spKAA%2BJsLbex6dZtBM9%2FGBS4z4PRK8fUCA6E%3D&reserved=0>, or mute the thread<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAL4WNRNNRLLPBT3AZ3WBHL3P5JDTLANCNFSM4HDUI7FQ&data=02%7C01%7Cdafishma%40microsoft.com%7C08b62f7cb26e46c8fef308d6fe51438f%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636976022311952893&sdata=pKWbHxZw3P5rIKXvHi%2F4jOf6jolB9GyCUZXL6%2FasKE0%3D&reserved=0>.

erich-wang commented:
@mikreite ,  could you address change requests from @chlahav ? Please let us know if the PR is ready for merge.

mikreite commented:
@erich-wang thanks for your reminder. There are still changes to be made, we are still not ready for merge. ETA for handling them by the next 2-3 weeks.

yungezz commented:
Hi @mikreite is it ok to close the PR now since you're not ready for merge now, and it's over 4 months, there's conflict already with latest code base. pls create new PR when you're ready for merge.

mikreite commented:
Hi @yungezz this resource is part of a private preview feature that is exposed in our RP manifest. We have approved the manifest change by having this open PR. This feature is going to be public (preview) by end of Sep. so the PR is supposed to be merged soon. Regarding the conflicts - I'll update the PR on a weekly basis, until last finishes are done. Please let me know if we should act differently, thanks.

yungezz commented:
@mikreite it's ok if the PR is still active.

nirwinter committed:
Adding event hub action to playbook RP

mikreite committed:
merge from upstream/master

OmAbraAzure committed:
adding securityEmail action type
